### ISSUE TITLE
Problem testing structured Pro Worker

### ISSUE THREAD

**Issue by cam-carter** [sorentwo/oban#649]
I'm a bit stumped on this at the moment, and I'm sure I can circumnavigate it somehow, but it seemed too strange not to report.

### Environment

- Oban Version -> `2.11 & pro 0.10`
- PostgreSQL Version -> `12.9`
- Elixir & Erlang/OTP Versions (`elixir --version`) -> `1.12.3 & otp 24.1.7`

### Current Behavior

I'm having some issues testing the new `Oban.Pro.Worker`, specifically a `structured` worker. I'm using `import Oban.Pro.Testing` (couldn't find anything on this in the official docs, so thanks @milmazz for your recent post) and am having issues with `process_job/1`.

The current function definition of my worker's `process/1` fn is written as follows:
```elixir
@impl true
def process(%Oban.Job{args: %__MODULE__{key: value})
```
I wrote this under the assumption that the following from the pro docs was true:
> A notable benefit is that the args passed to process/1 are converted into a struct

However, when testing this worker I'm seeing some strange behavior. With the above function definition it passes only after I save a change to the worker, e.g. simply adding an `IO.inspect/1` and saving the file results in this assertion passing:
```elixir
assert {:ok, result} = process_job(MyWorker, %{key: "value"})
```
But every attempt after the initial pass results in the following failure:
```
** (FunctionClauseError) no function clause matching in MyWorker.process/1

The following arguments were given to MyWorker.process/1:
     
         # 1
         %Oban.Job{__meta__: #Ecto.Schema.Metadata<:built, "oban_jobs">, args: %{"key" => "value"}, attempt: 0, attempted_at: nil, attempted_by: nil, cancelled_at: nil, completed_at: nil, conf: nil, conflict?: false, discarded_at: nil, errors: [], id: nil, inserted_at: nil, max_attempts: 20, meta: %{}, priority: 0, queue: "default", replace: nil, scheduled_at: nil, state: "available", tags: [], unique: nil, unsaved_error: nil, worker: "MyWorker"}
     
     Attempted function clauses (showing 1 out of 1):
     
         def process(%Oban.Job{args: %MyWorker{key: value}})
```

### Expected Behavior

I expect the test for `process/1` to pass with the given arguments every time.

**Comment by sorentwo** [comment]
I suspect this is the same issue happening with the recorded flakiness from #648. 

Looking into it tomorrow.

**Comment by sorentwo** [comment]
I'm not able to recreate this consistently, but I have recreated it a couple of times. So I'd appreciate any insight or tips into _how_ to recreate the problem.

**Comment by cam-carter** [comment]
> I'm not able to recreate this consistently, but I have recreated it a couple of times. So I'd appreciate any insight or tips into _how_ to recreate the problem—is there anything you're doing between test runs?

I simply just added a line in the test (`foo = "bar"`), saved, then re-run. The first run of the test after I saved the file passes, then every subsequent run fails.

**Comment by cam-carter** [comment]
> I suspect this is the same issue happening with the recorded flakiness from #648.
> 
> Looking into it tomorrow.

I was having issues with `recorded`, as well, but I wasn't sure exactly what it was. That was also in testing, didn't try anything in my dev environment. If you want more details I can open a new issue.

**Comment by cam-carter** [comment]
I'm definitely very eager to try out the new Pro Workers! The new changes look awesome! Thanks for your hard work 😄

**Comment by sorentwo** [comment]
Fix in Pro main and I'll have a fix out shortly. I was eventually able to recreate and craft a workaround—quite bizarre.

When opts are inspected, the stages all show up, and we have the expected meta:

```elixir
test "inspect __opts__" do
 IO.inspect BotCleaner.__opts__()
 IO.inspect BotCleaner.new(%{domain: "hold"}).changes
end

[
  stages: [
    {Oban.Pro.Stages.Standard, []},
    {Oban.Pro.Stages.Recorded,
     %Oban.Pro.Stages.Recorded{limit: 32000, to: :return}},
    {Oban.Pro.Stages.Structured,
     %Oban.Pro.Stages.Structured{
       keys: #MapSet<["domain", "user_agent"]>,
       required: #MapSet<[]>,
       worker: Oban.Workers.BotCleaner
     }}
  ],
  worker: "Oban.Workers.BotCleaner",
  queue: :default,
  max_attempts: 5
]
%{
  args: %{domain: "hold"},
  max_attempts: 5,
  meta: %{"recorded" => true, "structured" => true},
  worker: "Oban.Workers.BotCleaner"
}
```

When opts are called, but not inspected, the meta isn't applied:

```elixir
test "without inspect __opts__" do
 BotCleaner.__opts__()
 IO.inspect BotCleaner.new(%{domain: "hold"}).changes
end

%{args: %{domain: "hold"}, max_attempts: 5, worker: "Oban.Workers.BotCleaner"}
```

**Comment by sorentwo** [comment]
🚀 This is fixed in the just-published Pro v0.10.2
